packages/core/src/discovery.js

```diff
@@ -88,21 +88,52 @@ function waitForDiscoveryNetworkIdle(page, options) {
 
 // Creates an initial resource map for a snapshot containing serialized DOM
 function parseDomResources({ url, domSnapshot }) {
-  if (!domSnapshot) return new Map();
-  let isHTML = typeof domSnapshot === 'string';
-  let { html, resources = [] } = isHTML ? { html: domSnapshot } : domSnapshot;
-  let rootResource = createRootResource(url, html);
+  const map = new Map();
+  if (!domSnapshot) return map;
+  let allRootResources = new Set();
+  let allResources = new Set();
+
+  if (!Array.isArray(domSnapshot)) {
+    domSnapshot = [{ domSnapshot }];
```

Comment:
should this be an object array with domSnapshot key i it ? that doesnt seem to match your schema

Reply:
![Screenshot 2024-09-19 at 11 06 58 PM](https://github.com/user-attachments/assets/216c17c3-1451-421d-b756-13630fec034a)
This is correct. check schema

Reply:
Any specific reason to put width and domSnapshot separately ? shouldnt width be part of it ? 

Especially considering you are defining it in schema separately but not passing it here on line 97 ?